evalml/tests/automl_tests/test_automl.py

```diff
@@ -2524,47 +2306,13 @@ def test_max_batches_works(
         max_iterations=None,
         max_batches=max_batches,
         optimize_thresholds=False,
-        ensembling=use_ensembling,
+        _automl_algorithm=_automl_algorithm,
```

Comment:
This was an interesting one. Originally `test_max_batches_works` was not actually testing `max_batches` behavior. It was erroneously using the `max_iterations` stopping condition (which is set when using ensembling with `IterativeAlgorithm`. So I refactored this test to properly test `max_batches` behavior which can be seen through  `assert automl._get_batch_number() == max_batches + 1` since our stopping condition is `curr_batch > max_batch`.

Reply:
The rest of the test was moved to `test_max_batches_num_pipelines_iterative` as it was testing the behavior of iterative + ensembling + max batches.